### ISSUE TITLE
Default PermissionStereotypes are used anywhere?

### ISSUE THREAD

**Issue by jersiovic** [OrchardCMS/OrchardCore#4037]
Method  IPermissionProvider.GetDefaultStereotypes() that all the PermissionProvider implements looks that is useful for nothing. I didn't find any place in the code referencing it. Am I wrong? 

Is there any plan of using it in the next future?

**Comment by sebastienros** [comment]
If it was not used the roles would not be created with their default permissions. Found it here:
https://github.com/OrchardCMS/OrchardCore/blob/d729a2048a4390380d99e7be9f7968ff8f8e9271/src/OrchardCore.Modules/OrchardCore.Roles/Services/RoleUpdater.cs#L90

**Comment by jersiovic** [comment]
Oh sorry thank you.
So, if I add new default permissions for an streotype I need to add code to migrations of the module adding also those permissions if the featured was installed previously. Cause this code is only called by IFeatureEventHandler.Installed

**Comment by jersiovic** [comment]
It would be nice I could get through dependency injection RoleUpdater on migrations for  calling RoleUpdater.AddDefaultRolesForFeatureAsync to avoid duplicated code.
Currently only way is requesting all IFeatureEventHandlers and looking for the one it is the RoleUpdater.
An alternative would be to invoke Installed feature event on migrations, but this could cause not desired side effects cause in fact it is not true that a feature has been installed.
Any idea to address it in the best way?

**Comment by sebastienros** [comment]
Correct, a migration could help. We just need to call the service to add specific permissions to specific roles, but also be careful that some permissions might have been enabled/disabled manually. So a call in a migration should only reflect changes that can't be opposite to the current state. For new permissions it's easy.
